ReactWindows/ReactNative/Bridge/ReactDelegateFactoryBase.cs

```diff
@@ -141,27 +143,43 @@ public Promise(ICallback resolve, ICallback reject)
                 _reject = reject;
             }
 
-            public void Reject(string reason)
+            public void Resolve(object value)
             {
-                if (_reject != null)
+                if (_resolve != null)
                 {
-                    _reject.Invoke(new Dictionary<string, string>
-                    {
-                        { "message", reason },
-                    });
+                    _resolve.Invoke(value);
                 }
             }
 
-            public void Reject(Exception exception)
+            public void Reject(string code, string message)
             {
-                Reject(exception.Message);
+                Reject(code, message, default(Exception));
             }
 
-            public void Resolve(object value)
+            public void Reject(string message)
             {
-                if (_resolve != null)
+                Reject(DefaultError, message, default(Exception));
+            }
+
+            public void Reject(string code, Exception e)
+            {
+                Reject(code, e.Message, e);
+            }
+
+            public void Reject(Exception e)
+            {
+                Reject(DefaultError, e.Message, e);
+            }
+
+            public void Reject(string code, string message, Exception e)
```

Comment:
https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/bridge/PromiseImpl.java#L59

Just following the same convention, at some point we want to include the stack trace I think.